tools/cloud_storage_manifest_patcher/main.py

```diff
@@ -0,0 +1,393 @@
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+from time import sleep
+from functools import wraps
+from collections import namedtuple
+import time
+import datetime
+from typing import Union
+import json
+
+
+class SlowDown(Exception):
+    pass
+
+
+S3ObjectMetadata = namedtuple('S3ObjectMetadata',
```

Comment:
Typed version of namedtuple is an improvement. Example [here](https://github.com/redpanda-data/redpanda/blob/9e9db8e0ee259775f03ff9667a0310959ed31c81/tests/rptest/utils/expect_rate.py#L12-L20).